File: src/main/java/br/com/rotaractsorocabaleste/rotasales/item/ItemServiceImpl.java
package br.com.rotaractsorocabaleste.rotasales.item;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.Set;
import java.util.UUID;

@Service
@RequiredArgsConstructor
public class ItemServiceImpl implements ItemService {

    private final ItemRepository itemRepository;

    @Override
    public Item create(final Item item) {
        return itemRepository.save(item);
    }

    @Override
    public boolean belongsToEvent(final UUID id,
                                  final UUID eventId) {
        return itemRepository.findByIdAndEventId(id, eventId).isPresent();
    }

    @Override
    public Set<Item> findItemsByEventId(final UUID eventId) {
        return itemRepository.findByEventId(eventId);
    }

}

File: src/main/kotlin/br/com/rotaractsorocabaleste/rotasales/core/service/impl/ItemServiceImpl.kt
package br.com.rotaractsorocabaleste.rotasales.core.service.impl

import br.com.rotaractsorocabaleste.rotasales.core.entity.Item
import br.com.rotaractsorocabaleste.rotasales.core.service.ItemService
import br.com.rotaractsorocabaleste.rotasales.dataprovider.repository.ItemRepository
import org.springframework.stereotype.Service
import java.util.*

@Service
class ItemServiceImpl(
    private val itemRepository: ItemRepository
) : ItemService {
    override fun save(item: Item): Item = itemRepository.save(item)

    override fun findByEventId(eventId: UUID): List<Item> = itemRepository.findByEventId(eventId)
}